src/Umbraco.Forms.Integrations.Commerce.EMerchantPay/PaymentProviderWorkflow.cs

```diff
@@ -138,18 +153,29 @@ public override WorkflowExecutionStatus Execute(Record record, RecordEventArgs e
             var formId = record.Form;
             var recordUniqueId = record.UniqueId;
 #endif
-            var uniqueIdKey = mappings.First(p => p.CustomerProperty == nameof(MappingValues.UniqueId)).Field.Id;
-            var statusKey = mappings.First(p => p.CustomerProperty == nameof(MappingValues.Status)).Field.Id;
+            var uniqueIdKey = UniqueId;  
+            var statusKey = RecordStatus;
+
+
+            var numberOfItems = string.IsNullOrEmpty(NumberOfItems)
+                ? 0
+#if NETCOREAPP
+                : int.Parse(context.Record.RecordFields[Guid.Parse(NumberOfItems)].ValuesAsString());
+#else
+                : int.Parse(record.RecordFields[Guid.Parse(NumberOfItems)].ValuesAsString());
+#endif
 
             var payment = new PaymentDto
             {
                 TransactionId = transactionId.ToString(),
                 Usage = _paymentProviderSettings.Usage,
-                NotificationUrl = $"{_paymentProviderSettings.UmbracoBaseUrl}/umbraco/api/paymentprovider/notifypayment?formId={formId}&recordUniqueId={recordUniqueId}&statusFieldId={statusKey}",
+                NotificationUrl = $"{_paymentProviderSettings.UmbracoBaseUrl}umbraco/api/paymentprovider/notifypayment?formId={formId}&recordUniqueId={recordUniqueId}&statusFieldId={statusKey}",
                 ReturnSuccessUrl = _urlHelper.GetPageUrl(int.Parse(SuccessUrl)),
                 ReturnFailureUrl = _urlHelper.GetPageUrl(int.Parse(FailureUrl)),
                 ReturnCancelUrl = _urlHelper.GetPageUrl(int.Parse(CancelUrl)),
-                Amount = int.Parse(Amount),
+                Amount = numberOfItems != 0
```

Comment:
We should support decimal values here I think - could well be the amount is 99.99 euros/pounds.  So `decimal.Parse` instead of `int.Parse`.

Reply:
Their API expects an integer value, the decimal point is removed, but on their side they are considering the last 2 digits as decimals and automatically convert.
For example, a value of 1500 will be converted to 15.00, 2050 to 20.50 and so on.